### ISSUE TITLE
Feature: Create mesa-models PyPI package 

### ISSUE THREAD

**Issue by tpike3** [projectmesa/mesa-examples#54]
Per Mesa [#1707](https://github.com/projectmesa/mesa/issues/1707#issuecomment-1637909901) we need to create mesa-model PyPI library for easy use access to models.

**Comment by EwoutH** [comment]
We also should tag releases in sync with the main repo, so that the mesa examples are guaranteed to work with a certain mesa version.

For examples, if we tag 2.2.0 of Mesa, we should test and update all the Mesa examples for that version, and then also tag 2.2.0 for the Mesa examples.

**Comment by Corvince** [comment]
Good idea, but I don't think we should be completely in sync. What happens when we add more examples or need a Bugfix for some example? If we then push the mesa-examples version, do we also push mesa to a higher version? This doesn't make sense, so we should probably only do it for Mayor.Minor versions and abuse the Bugfix version for updates to mesa examples. It won't be semver for Mesa examples, but semver doesn't make much sense for examples in the first place, so that should be fine

We could probably do this automatically when we release a new version of mesa via GitHub actions. But of course only if the models work on the new version.

**Comment by EwoutH** [comment]
We can just add an extra digit right? Like `major.minor.patch.build`.

Actually, I think PEP 440 [post releases](https://peps.python.org/pep-0440/#post-releases) are perfect for this. Like `2.2.0.post1`.

**Comment by EwoutH** [comment]
I will try to get this working upcoming week.

**Comment by EwoutH** [comment]
> We also should tag releases in sync with the main repo, so that the mesa examples are guaranteed to work with a certain mesa version.

@projectmesa/maintainers I still think this is an idea worth perusing. Especially with the 3.0 changes incoming, it would be really nice to have an 2.3 branch with 2.3 tags, on which we can maintain examples, and a main branch on which we can develop new examples for 3.0.

**Comment by wang-boyu** [comment]
Do we need separate branches for each tag, or is it possible to have only one main branch with all tags, similar to how packages are released?

**Comment by EwoutH** [comment]
I think for breaking changes we need to have branches. So probably `mesa-2.x`, `mesa-3.x`, etc
